client/trino-client/src/main/java/io/trino/client/auth/external/HttpTokenPoller.java

```diff
@@ -46,11 +48,24 @@
     private static final String USER_AGENT_VALUE = "TrinoTokenPoller/" +
             firstNonNull(HttpTokenPoller.class.getPackage().getImplementationVersion(), "unknown");
 
-    private final OkHttpClient client;
+    private final Supplier<OkHttpClient> client;
 
     public HttpTokenPoller(OkHttpClient client)
     {
-        this.client = requireNonNull(client, "client is null");
+        requireNonNull(client, "client is null");
+        this.client = () -> client;
+    }
+
+    public HttpTokenPoller(OkHttpClient client, Consumer<OkHttpClient.Builder> refreshableClientConfig)
```

Comment:
This method is unused in this commit, so please move to commit where it is used.